### PR TITLE
node: Call signatures for gc()

### PR DESCRIPTION
**Pull request by rotu**
Previously, gc did not have correct call signatures. Now it does.

Taken from https://github.com/nodejs/node/blob/v20.x/deps/v8/src/extensions/gc-extension.h

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/node/test/globals.ts
1. types/node/globals.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a12e386880c5f0e0009943aab9b54ca5a22f114a
</p>

File: types/node/globals.d.ts [hunk 1]
```diff
@@ -53,10 +53,34 @@ declare global {
...
 
+    interface GCFunction {
+        (options: {
+            execution?: "sync";
+            flavor?: "regular" | "last-resort";
+            type?: "major-snapshot" | "major" | "minor";
+            filename?: string;
+        }): void;
+        (options: {
+            execution?: "async";
```
Comment (Renegade334) [inline types/node/globals.d.ts:64]:
```suggestion
            execution: "async";
```
so as not to overlap with the sync signature.

File: types/node/globals.d.ts [hunk 2]
```diff
@@ -53,10 +53,34 @@ declare global {
...
+        }): Promise<void>;
+        (options?: boolean): void;
+        (
+            options?: {
+                execution?: "async" | "sync";
+                flavor?: "regular" | "last-resort";
+                type?: "major-snapshot" | "major" | "minor";
+                filename?: string;
+            } | boolean,
+        ): void | Promise<void>;
```
Comment (Renegade334) [inline types/node/globals.d.ts:77]:
```suggestion
```
Looks like an implementation signature, which isn't required here.

**Review by Renegade334 — CHANGES_REQUESTED** [review]

File: types/node/globals.d.ts (hunk 2 above)
Comment (rotu) [reply types/node/globals.d.ts]: I'm not sure if it's required or not. If I omit it, then this code fails to typecheck:

```tsx
function foo(async?:boolean){
  return gc({
    execution: async ? "async":"sync"
  })
}
```

[Playground Link](https://www.typescriptlang.org/play/?#code/JYOwLgpgTgZghgYwgAgOIGEBiBXECzAD2IyA3gLABQyNyAFIQA4HEDOAXGVbT8hAB4QE2FiAD8nAESsAnnkkBubrxowANnABuhKBOSSoEAObYNUScgA++jazABaQ6x1hFylWBmMIeyQFs4ACsde1YQOEZWAAtCVyt9AODzeP9QHTdqFWQYYDUIcL8fTjsoUCMlTJoAXwBKTm1gABMKngZmIhAOLkreASERDqk4WXkWrPUtHV9DEzMLa0lbBycXDKzkT29fRJCwiOjY+YSg9JS-NPMxlRy8gqLkErKr5FrOAAUoQnPWCAAeBsaAD5nm1RKw9AAjQiEPJwEB1ZAAipVKhURpCMwoBBsMDIIwIdgYHB4USoygwXD4DrZaF0YZyBASKEwiBwmoUTKGMDYKAkfF0Dk8PrCUSc

https://github.com/microsoft/TypeScript/issues/14107

Comment (Renegade334) [reply types/node/globals.d.ts]: Not sure what the usage case would be, but if you want to allow `execution` to be a union while keeping the return type dependent on its value, you'd need to use a conditional return. I guess that would look something like:

```ts
    interface GCFunction {
        <ExecutionOption extends "sync" | "async" | undefined>(options: {
            execution: ExecutionOption;
            flavor?: "regular" | "last-resort" | undefined;
            type?: "major-snapshot" | "major" | "minor" | undefined;
            filename?: string | undefined;
        }): ExecutionOption extends "async" ? Promise<void> : void;
        (options: {
            flavor?: "regular" | "last-resort" | undefined;
            type?: "major-snapshot" | "major" | "minor" | undefined;
            filename?: string | undefined;
        }): void;
        (options?: boolean): void;
    }
```

Comment (Renegade334) [reply types/node/globals.d.ts]: Aye, this wouldn't allow a union type for `execution`, although I don't really see the usage case.

If you wanted this, you'd need a conditional return, something like:

```ts
    interface GCFunction {
        <ExecutionOption extends "sync" | "async" | undefined>(options: {
            execution: ExecutionOption;
            flavor?: "regular" | "last-resort" | undefined;
            type?: "major-snapshot" | "major" | "minor" | undefined;
            filename?: string | undefined;
        }): ExecutionOption extends "async" ? Promise<void> : void;
        (options: {
            flavor?: "regular" | "last-resort" | undefined;
            type?: "major-snapshot" | "major" | "minor" | undefined;
            filename?: string | undefined;
        }): void;
        (options?: boolean): void;
    }
```

Comment (rotu) [reply types/node/globals.d.ts]: Okay. I'm happy enough with the current implementation and the conditional return type is ugly.

It seriously bugs me that `ReturnType<typeof gc>` and `Parameters<typeof gc>` don't respect the overloading. But I'm going to chalk this up to buggy behavior of TypeScript itself and flog on the upstream issues.

File: types/node/globals.d.ts [hunk 3]
```diff
@@ -53,10 +53,26 @@ declare global {
...
+            flavor?: "regular" | "last-resort";
+            type?: "major-snapshot" | "major" | "minor";
+            filename?: string;
+        }): void;
+        (options: {
+            execution: "async";
+            flavor?: "regular" | "last-resort";
+            type?: "major-snapshot" | "major" | "minor";
+            filename?: string;
+        }): Promise<void>;
```
Comment (Renegade334) [inline types/node/globals.d.ts:68]:
```suggestion
        (options: {
            execution?: "sync" | undefined;
            flavor?: "regular" | "last-resort" | undefined;
            type?: "major-snapshot" | "major" | "minor" | undefined;
            filename?: string | undefined;
        }): void;
        (options: {
            execution: "async";
            flavor?: "regular" | "last-resort" | undefined;
            type?: "major-snapshot" | "major" | "minor" | undefined;
            filename?: string | undefined;
        }): Promise<void>;
```
Apologies for not including previously - this is required for compatibility with `exactOptionalPropertyTypes`.

**Review by Renegade334 — CHANGES_REQUESTED** [review]

File: types/node/globals.d.ts (hunk 3 above)
Comment (rotu) [reply types/node/globals.d.ts:68]: > Apologies for not including previously - this is required for compatibility with `exactOptionalPropertyTypes`.

How so? Why should `undefined` be accepted?

My opinion is that it would be wrong - if the user actually has `exactOptionalPropertyTypes` set, they regard `undefined`-valued properties as a mistake, and we should respect that.

For background, the value `undefined` is [treated internally](https://github.com/nodejs/node/blob/d65b17082bd6b559fc6666e14ef5bd0079389ecc/deps/v8/src/extensions/gc-extension.cc#L54) the same as passing a property that is not a string. That is, for all arguments, it's tolerated but ignored, just as if passing `null` or even `42`.

Comment (Renegade334) [reply types/node/globals.d.ts:68]: Indeed, and it's the convention in this repo to type options interfaces in that way ([example](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b4ae44884a5a4a56d6059d81f1567d3f6983dbe8/types/node/zlib.d.ts#L96-L123)) to allow for "positively" specifying the default parameter.

Comment (rotu) [reply types/node/globals.d.ts:68]: It seems like that was a change to preserve backwards compatibility, but since this declaration did not precede TS 4.4, there is no backwards compatibility to maintain.

It makes the code less correct and less readable. If you want to make that change in another PR, I can't stop you.

https://github.com/microsoft/TypeScript/issues/44419#issuecomment-874662681

> The key with these automated PRs is that they are 'backwards' compatible in the sense that 4.4 behavior with exactOptionalPropertyTypes enabled is gives you the same type system behavior as 4.3 without it.  ... However, post the PRs it now easy to switch the semantics to be more accurate on a case by case basis.

Comment (Renegade334) [reply types/node/globals.d.ts:68]: I believe the meta is still to encode `undefined` as the semantic value for "default" (allows for overriding to default in merged objects etc.) but I don't want to block this from being reviewed.

**Review by Renegade334 — APPROVED** [review]
Comments for onward review:
- The `flavor` and `filename` options, as well as the `type` option `"major-snapshot"`, aren't present in the V8 extension prior to v22. These are silently ignored in earlier versions, so having them in the v20 types wouldn't necessarily break anything.
- [Discussion](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69821/files#r1659045134) on options interface properties and `undefined`.

File: types/node/globals.d.ts [hunk 4]
```diff
@@ -53,10 +53,26 @@ declare global {
     // Same as module.exports
     var exports: any;
 
+    interface GCFunction {
+        (options: {
+            execution?: "sync";
+            flavor?: "regular" | "last-resort";
+            type?: "major-snapshot" | "major" | "minor";
+            filename?: string;
```
Comment (jakebailey) [inline types/node/globals.d.ts:61]: I highly suspect that these should all have `| undefined` to support `exactOptionalPropertyTypes`.

See: [Playground Link](https://www.typescriptlang.org/play?exactOptionalPropertyTypes=true&ts=4.4.2#code/PTAEHUEsBcAtQKYA8CGBjaB5ADtSB7AOxQBsAFAJ32wQugE8AVemgZ0WICMSEATAGlDMaAZTQVIuALAAoEKFgp2raBIy1QFAK492KKlsK9QAd0XRQ9fFtBoUhUKwQX7oaniKlZ87FRp16UGh8ADpZbzAAMXwKRFQAW2weQThIdkhCaFoAM3QEUF4ENBJ9BHY4FAs4Wny00BQI0F9qWgZTWEg0eDsHTnyifuygk3xHVQyAc1YALlAAcl59AGs5t1i5kkgJ2Gg5xpjQGEdYaxJjQnwLPsOHardOACsi6DCZWQysily0fIBVJwoABEELkdNB2ABvWSgUDyRiwfIoThOQg-NxDFCgABupC0+QoCF8ZQQmXYc1Y9BUCHiexkMLQ+BIMXh1IQmCxtAkhQA-LMAESLChLPmgAA+oD5m220D5AG5ZABfcJyMBQODWKoddjZEoTDhIngCIIIgn1U1wAn5HEkPHsMydeBWGw9RpOCwAAwZTIoLPibI5FC5CHdQVGfX5guFgklWx2IvsxndhkK2QyfHdr0aImceEIeru1rxodASaMILTvBDJkgJBI9Vr+BMoHi+BUoAAUigcWIJLhNIY8H7Gl0ikttQc7sg0llUflgqBcjXBPauod2CDss9IBySIFciQKa95AAJRsIAMpE0Icn1BdaABe97TrHKo1S7D6iixBAoISECIKctCBgAhCEabIDk9RlmQRP12U5SBChDAB9Xh8DKQg5mgZCbmKLRChLZMgPTTMVX-fIdRQPV3WQdAsFwUDSEoFoAmEMoQ3iFAljKIQWAQHtJBcNA0C0ChKgQXdGk46BVzuFMMhAogmiob9CmMJR6gcdxGLrZp-AYUjCmKUo71RDwHAmZx-lobNoFzKYAAoAEpZmsoEQRQMFWHlN4ZAZQg2zdez2AAXlASzoDc2zgucnygsmVgQi9GDWXgwNEPyMKBWWOVZHivNEuSn1YP9BCCKyqU4x8xoACEtAsC5oGmPKcwSpLoOK1KAyDUAwqI1NCD4aqyMoH9izuSjqNojAcHMpi-FaJg+NYd1LzqEwNMappW1YSBuAQEIgA)

File: types/node/globals.d.ts (hunk 3 above)
Comment (jakebailey) [reply types/node/globals.d.ts:68]: Oops, didn't see this thread.

You definitely strongly recommend putting `| undefined` on these. You have the backwards compat flipped; this is for 4.4 and _newer_. If you use this API today and pass in an undefined param (maybe you want to conditionally do something), it won't work.

Comment (rotu) [reply types/node/globals.d.ts:68]: No, I think I understand the compatibility reason.

AFAICT, the compatibility reason is so that code which worked pre-4.4 with the old behavior would still pass type-checking post-4.4.

It’s not needed since the declaration did not exist before Typescript introduced the breaking change to the `?` modifier.

For merged objects, `undefined` behaves very differently from function calls, so it’s better to *not* confuse the two at the type level all else being equal.

This will print 1 `((x=1)=>{console.log(x)})(undefined)`

This will print `undefined`: `console.log(Object.assign({x:1}, {x:undefined}).x)`

Comment (jakebailey) [reply types/node/globals.d.ts:68]: Without `| undefined`, this code is not legal:

```
global.gc({ execution: isSync : "sync" : undefined })
```

Comment (rotu) [reply types/node/globals.d.ts:68]: > Without `| undefined`, this code is not legal:
> 
> ```
> global.gc({ execution: isSync : "sync" : undefined })
> ```

Nor should it be!

1. Union types don't distribute over overloaded function calls so this won't work anyway (assuming you meant `global.gc({ execution: isAsync ? "async" : undefined })`)

2. If you pass `global.gc({})` or `global.gc({execution:undefined})`, it will default to the legacy "truthy parameter" behavior of a *minor*, synchronous GC pass. So you *don't* want conditionally-undefined properties for this API.

3. `| undefined` is wordier. Better to keep things short and sweet.

File: types/node/globals.d.ts (hunk 4 above)
Comment (rotu) [reply types/node/globals.d.ts:61]: Addressed in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69821#discussion_r1678715212

File: types/node/globals.d.ts (hunk 3 above)
Comment (jakebailey) [reply types/node/globals.d.ts:68]: I chose a bad property, clearly; let me try again: [Playground Link](https://www.typescriptlang.org/play/?exactOptionalPropertyTypes=true#code/PTAEGUHcEMAdQKYA9oGMAuB5W6CWB7AO2gBsAFAJ31gQvQE8AVemgZ1ACMF1IEFDQ6CgFcEoaIQAmoAGalWCAFwAoZSFAABZGiw4CxclRp0mLBK0WCRCVbkLpac1GIDiAYQBiwwhn2gA3sqgwaAAFNR4RBYBQSFxyAiowpGEAPyWAESs9D4ZANyxccEyJNAAbvgU6aAZFAgA5sKlFBmgAD41pazoALR1rJXo+YVFDDTVGQC20ABWlT2sxLCsABb4Q+0103Mtm1N2lcNFITK4JPzQkwjV3RR29QVxAL4AlJYVuJKPIeF6UZaBY7BBJJFKZaDZXLfY4lcqVCZ1RrNVodDJdXr9QZHIFja6ZbbzRZwVbrFFbWaHPaTA4taFFU7nYhXG5Ce5016WSj4akKAA8H0kAD4Ck9VJJEs0xGVoBRQPVUJZ3F4fCkCsoZN5fERQJJ8C5UKFcKwAEoNJoyywcfD4c4SAA0sjOFyullu9xeMTi8tCgKBsIqFEsRtNSJloFSNUR5t2lm84tOhAQkjtIziDOdCBTzxeyieQA)



> 3\. `| undefined` is wordier. Better to keep things short and sweet.

Wordiness or not, adding it changes the meaning.

Comment (rotu) [reply types/node/globals.d.ts:68]: > I chose a bad property, clearly; let me try again

Since "regular" is the default, I expect you mean to toggle between flavor="regular" vs "last-resort". And that filename is not intended since this only has an effect for type="major-with-snapshot".

So I'll respond as if you have written this:
```ts
function doGc(isLastResort: boolean) {
    gc({
        flavor: isLastResort ? "last-resort" : undefined,
    })
}
```

Because of how this sets `found_options_object` in [gc-extension.cc](https://github.com/nodejs/node/blob/3a5b5428c6509f91306e10486f1e56e3b1e80195/deps/v8/src/extensions/gc-extension.cc), calling `doGc(true)` would result in a major, last-resort gc pass. But `doGc(false)` would result in a MINOR gc pass, which is not what's intended.

You should instead write:
```ts
function doGc(isLastResort: boolean) {
    gc({
        flavor: isLastResort ? "last-resort" : "regular",
    })
}
```
so that `doGc(false)` will result in a MAJOR gc pass, as probably intended.

---

> Wordiness or not, adding it changes the meaning.

100% agreed. Which is why it shouldn't be added - it makes the user has to ask "is there a difference between these two possible inputs"? It's also why I didn't add `| null | "" | 0` to every property, even though these result in the exact same behavior that `undefined` would.

If the user wants to be strict about "exactOptionalPropertyTypes" (which IMO is very reasonable), then blindly adding `| undefined` to all APIs means that TypeScript *won't* catch accidentally passing `undefined` to the property. It defeats the of having such a check in the first place!

Comment (jakebailey) [reply types/node/globals.d.ts:68]: I am really not in a position to make claims about what particular inputs do or whether or not my specific code "makes sense"; I am more concerned about the types and ensuring DT consumers get working types. I can just as easily produce an example that doesn't use `gc` to avoid getting in the weeds about this particular API.

> If the user wants to be strict about "exactOptionalPropertyTypes" (which IMO is very reasonable), then blindly adding `| undefined` to all APIs means that TypeScript _won't_ catch accidentally passing `undefined` to the property. It defeats the of having such a check in the first place!

The difference is just about whether `gc({})` and `gc({ flavor: undefined })` behave the same way. If they do, then `| undefined` should be there. I'm not sure how I can explain this more clearly.

There are some APIs where this does matter. It's not clear that it does for `gc`.

Comment (rotu) [reply types/node/globals.d.ts:68]: > I am more concerned about the types and ensuring DT consumers get working types.

What do you mean by "working types"? This type works as intended. The reason to add `| undefined` was so that transitioning to 4.4 wouldn't introduce new breakages for consumers. Not a concern for this PR, since TS 4.4 is already out there.

> The difference is just about whether `gc({})` and `gc({ flavor: undefined })` behave the same way. If they do, then `| undefined` should be there. I'm not sure how I can explain this more clearly.

`gc({})`, `gc(1)`, `gc({ flavor: undefined })`, `gc({ flavor: null })`, `gc({ flavor: "vanilla" })` *all* behave the same way; if I could prohibit them all at the type level (in an idiomatic and readable way), I would.

> There are some APIs where this does matter. It's not clear that it does for `gc`.

Agreed. That's why I omitted it - same as the reason I omitted other values that add no expressive power to the API. Even `null` which has the analogous "strictNullChecks" option.

----

A better pattern for defining a conditional property on an object literal which DOESN'T require the consuming function to weaken its type signature:

```ts
someFunction({
    key: value,
    ... (condition1 ? {key1: value1} : {key2: value2}),
    ... (condition2 && {key3: value3}),
})
```
----

Why do you think explicitly allowing `undefined` as a property value is a *good* idea?

Comment (jakebailey) [reply types/node/globals.d.ts:68]: > The reason to add `| undefined` was so that transitioning to 4.4 wouldn't introduce new breakages for consumers. Not a concern for this PR, since TS 4.4 is already out there.

This continues to be backwards. This is all about users of the flag.



> Agreed. That's why I omitted it - same as the reason I omitted other values that add no expressive power to the API.

You've flipped the meaning of my statement; the fact that it doesn't matter is a reason _to_ include `| undefined`.

> Even `null` which has the analogous "strictNullChecks" option.

That's not what that option means... That flag controls the existence of `undefined` and `null` both.



> Why do you think explicitly allowing `undefined` as a property value is a _good_ idea?

Because that pattern is not on its face better; needing to control the exact shape of an options bag when the API simply does not care about it is precisely _why_ `| undefined` is added to people's types.

Types are never "always right" or "always wrong", it's a scale of usibility. Plenty of code out there could accept garbage and "behave" in a defined way, but the types make the choice to disallow constructs that are probably undesirable, passing in `1`, `"vanilla"`, etc, as you've described. I simply do not think that disallowing the passing of `| undefined` is the right place to draw the line.

You seem impossible to convince of this, though, so I'll just drop it. This package is the most critical DT package, hence trying to ensure that the right changes are made.

Comment (rotu) [reply types/node/globals.d.ts:68]: > You've flipped the meaning of my statement; the fact that it doesn't matter is a reason _to_ include `| undefined`.

By adding `| undefined` to signatures, you're encouraging people to be sloppy when it adds no expressiveness.

>> Even `null` which has the analogous "strictNullChecks" option.

> That's not what that option means... That flag controls the existence of undefined and null both.

The same justification for adding `| undefined` suggests adding `| null`.
strictNullChecks allows sloppy code in the same way

```ts
gc({
  flavor: isLastResort ? "last-resort" : null,
})
```

[Playground link](https://www.typescriptlang.org/play/?strictNullChecks=false&exactOptionalPropertyTypes=true#code/PTAEGUHcEMAdQM4BcBOBLAxkgcgVwDb4DCAFgKYYDWCoARmUpGWQHaiq5mjQsAmoAM2j4EZAFwAoCSFAABZOix5CpCtTGDhoqWhZIyKIRi4BxIgDFcLLGgD2bAN4TQL0AApbsJHZYINT10DQMgAPClxvewB+DQAiBABPa1iAbmcglwF8aAA3WxQY0FiUMgBzAmgUWNAAHyLs5ABaEoR8pFT0jKQE2DJC2IBbaAArfMaEFjgEElt22qKh0ar5wd18jozMtHxWaAG+jQVdUrTAgF8ASg08tF5T1w8vHz9QAM3Q8MiWOOhE5PuMllcvl+iVytllnVYg0kM0yK0UO0AUFur1+osxhMpjM5lCMZCFmsqsjAgJtrt9oUjiwTp1QJcNAAFFC2AZoUQAHhuvAAfGkzlJeBQIVwcpVQKUMBozJZrF80hIBFYbPZQLxbCYMG52QAZX5IABK8LaGlotlsOx4ABpBOTJvtDqhjhdXnTJW43oDsnkUBpdfqjQikKAovV9XCg9UNCwCPgrXTLhIzkA)

>  I simply do not think that disallowing the passing of `| undefined` is the right place to draw the line.

I'm not "disallowing" passing `| undefined`. I'm just not adding it since there's little benefit in readability or usability.

**Comment by rotu** [comment]
> I've tried to search Node's docs for info on these options, but none exist. Is this really an API that should be documented? Are these options stable? Or are they specific to the V8 version included in Node and can change at any time?

Here's what it says about the `--expose-gc` flag to enable this API:

https://github.com/nodejs/node/blob/f09063752b6e290f5cc32069e7fd4e89a3860015/doc/api/cli.md#--expose-gc

> Stability: 1 - Experimental. This flag is inherited from V8 and is subject to change upstream.

Extremely useful for testing, but if there's a better place for this, I'd welcome it. I'm a little surprised it lives in Node's global namespace instead of in the `node:v8` module.

**Comment by jakebailey** [comment]
The only last thing to consider is whether these changes should also apply to the v16 and v18 directories.

**Review by sheetalkamat — DISMISSED** [review]
pls change v18 as mentioned.

**Comment by jakebailey** [comment]
Unfortunately this PR wasn't merged before v22 was added, so there's now a v20 dir that needs to be updated too. I'd merge from master first and fix that.

File: types/node/v18/globals.d.ts [hunk 5]
```diff
@@ -52,10 +52,22 @@ declare global {
...
     var exports: any;
 
+    interface GCFunction {
+        (options: {
+            execution?: "sync";
+            type?: "major-snapshot" | "major" | "minor";
+        }): void;
+        (options: {
+            execution: "async";
+            type?: "major-snapshot" | "major" | "minor";
```
Comment (Renegade334) [inline types/node/v18/globals.d.ts:62]:
```suggestion
            type?: "major" | "minor";
```

File: types/node/v18/globals.d.ts [hunk 6]
```diff
@@ -52,10 +52,22 @@ declare global {
     // Same as module.exports
     var exports: any;
 
+    interface GCFunction {
+        (options: {
+            execution?: "sync";
+            type?: "major-snapshot" | "major" | "minor";
```
Comment (Renegade334) [inline types/node/v18/globals.d.ts:58]:
```suggestion
            type?: "major" | "minor";
```

**Review by Renegade334 — CHANGES_REQUESTED** [review]
The major-snapshot value here is v22-only, so this will just need a tweak. This change should also apply to the copy of the definitions added to v20 (as per [comment](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69821#issuecomment-2276353292)).

**Comment by jakebailey** [comment]
I reopened this with the idea that it'd be updated for v16, but I'm not going to block on that.

Any remaining concerns here?

**Comment by rotu** [comment]
Well it seems to have been closed because the CI failed, but I don't think that was due to my changes. At any rate, I think there haven't been any concerns since last commit.

**Review by jakebailey — APPROVED** [review]

**Comment by rotu** [comment]
Ready to merge
